netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -205,6 +209,18 @@ private void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
     }
   }
 
+  private String getOrUpdateAuthority(AsciiString authority) {
+    if (authority == null) {
+      return null;
+    } else if (!authority.equals(lastKnownAuthority)) {
+      lastKnownAuthority = authority;
+    }
+
+    // AsciiString.toString() is internally cached, so subsequent calls will not
```

Comment:
This is not efficient enough.  If the client changes the authority every request and the server doesn't read from it, it will unnecessarily convert it.  Why not do the conversion when getAuthority is called?